### PR TITLE
refactor(cli): generate yargs configuration from TS

### PR DESCRIPTION
**Pull request by comcalvi**
### Issue # (if applicable)

N/A

### Reason for this change

We'd like to create a programmatic interface to the CDK Toolkit. A bonus of this overhaul is moving to a single source of truth for both the programmatic interface to the CDK Toolkit and the command line interface to the CDK Toolkit. This PR generates the existing `yargs` configuration from a TS configuration. In the long term, we'd generate the `yargs` configuration purely from the programmatic interface to the Toolkit, but this is an improvement with less effort. 

### Description of changes

Creates a new package, `@aws-cdk/yargs-gen`, which generates our `yargs` configuration from a `CliConfig` defined in `aws-cdk/config.ts` using `@cdklabs/typewriter`. 

### Description of how you validated changes

N/A yet. 

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mrgrain** [comment]
I'm a inclined the check the generated code in for visibility and reviewability. WDYT?

**Comment by comcalvi** [comment]
My first instinct was to follow the codegen pattern we have for L1s, which is to ignore them, but the nature of the `yargs` code is very different from that of the L1s, so I agree that checking this file in is good. I've added it.

File: packages/aws-cdk/bin/yargs-gen
Comment (mrgrain) [inline packages/aws-cdk/bin/yargs-gen:1]: `bin` is by convention for published binaries. This helper script should go into a `scripts` directory or similar.

File: packages/aws-cdk/lib/config.ts [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+          'outputs-file': { type: 'string', alias: 'O', desc: 'Path to file where stack outputs will be written as JSON', requiresArg: true },
+          'previous-parameters': { type: 'boolean', default: true, desc: 'Use previous values for existing parameters (you must specify all parameters on every deployment if this is disabled)' },
+          'toolkit-stack-name': { type: 'string', desc: 'The name of the existing CDK toolkit stack (only used for app using legacy synthesis)', requiresArg: true },
+          'progress': { type: 'string', choices: [StackActivityProgress.BAR, StackActivityProgress.EVENTS], desc: 'Display mode for stack activity events' },
+          'rollback': {
+            type: 'boolean',
+            desc: "Rollback stack to stable state on failure. Defaults to 'true', iterate more rapidly with --no-rollback or -R. " +
+              'Note: do **not** disable this flag for deployments with resource replacements, as that will always fail',
+          },
+          'R': {
```
Comment (mrgrain) [inline packages/aws-cdk/lib/config.ts:138]: This seems like we should roll it into the model for `rollback`.

"Negative" options are very much a cli only thing.

File: packages/aws-cdk/lib/config.ts [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+          'force': { alias: 'f', type: 'boolean', desc: 'Always deploy stack even if templates are identical', default: false },
+          'toolkit-stack-name': { type: 'string', desc: 'The name of the existing CDK toolkit stack (only used for app using legacy synthesis)', requiresArg: true },
+          'progress': { type: 'string', choices: [StackActivityProgress.BAR, StackActivityProgress.EVENTS], desc: 'Display mode for stack activity events' },
+          'rollback': {
+            type: 'boolean',
+            desc: "Rollback stack to stable state on failure. Defaults to 'true', iterate more rapidly with --no-rollback or -R. " +
+              'Note: do **not** disable this flag for deployments with resource replacements, as that will always fail',
+          },
+          // same hack for -R as above in 'deploy'
+          'R': {
```
Comment (mrgrain) [inline packages/aws-cdk/lib/config.ts:280]: Same as other

File: tools/@aws-cdk/yargs-gen/test/cli.test.ts
Comment (mrgrain) [inline tools/@aws-cdk/yargs-gen/test/cli.test.ts:1]: Let's have at least some small tests with a few common options.

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: packages/aws-cdk/bin/yargs-gen
Comment (mrgrain) [reply packages/aws-cdk/bin/yargs-gen:1]: Still not resolved.

Comment (mrgrain) [inline packages/aws-cdk/bin/yargs-gen:1]: bin is by convention for published binaries. This helper script should go into a scripts directory or similar.

**Review by mrgrain — APPROVED** [review]

**Pull request merged**